pmd-core/src/main/java/net/sourceforge/pmd/properties/PropertyBuilder.java

```diff
@@ -362,7 +363,7 @@ public RegexPropertyBuilder defaultValue(String pattern, int flags) {
         private final ValueParser<V> parser;
         private final Supplier<C> emptyCollSupplier;
         private final Class<V> type;
-        private char multiValueDelimiter;
+        private char multiValueDelimiter = MultiValuePropertyDescriptor.DEFAULT_DELIMITER;
```

Comment:
Regarding your comment about the delimiter, I set the default here, which is why it can be omitted in stringMultiProperty.